File: C++/Program/Komputer.cpp
#pragma once
#include <iostream>
#include <string>
#include <vector>
#include "Cpu.cpp"
#include "HDD.cpp"
#include "Ram.cpp"
#include "VGA.cpp"
#include "SSD.cpp"
#include "Keyboard.cpp"
#include "Mouse.cpp"
#include "Speaker.cpp"

using namespace std;

class Komputer
{
private:
    string nama;
    string merk;
    Cpu cpu;
    vector<Ram> ramList;
    HDD hdd;
    VGA vga;
    SSD ssd;
    Keyboard keyboard; 
    Mouse mouse;       
    Speaker speaker;  
public:
    Komputer()
    {
    }

    Komputer(string nama, string merk, Cpu cpu, vector<Ram> ramList, HDD hdd, VGA vga, SSD ssd, Keyboard keyboard, Mouse mouse, Speaker speaker)
    {
        this->nama = nama;
        this->merk = merk;
        this->cpu = cpu;
        this->ramList = ramList;
        this->hdd = hdd;
        this->vga = vga;
        this->ssd = ssd;
        this->keyboard = keyboard;
        this->mouse = mouse;
        this->speaker = speaker;
    }

    void setNama(string nama)
    {
        this->nama = nama;
    }

    void setCpu(Cpu cpu)
    {
        this->cpu = cpu;
    }

    void setRam(vector<Ram> ramList)
    {
        this->ramList = ramList;
    }

    void sethdd(HDD hdd)
    {
        this->hdd = hdd;
    }

    void setVga(VGA vga)
    {
        this->vga = vga;
    }

    void setSsd(SSD ssd)
    {
        this->ssd = ssd;
    }

    void setMerk(string merk)
    {
        this->merk = merk;
    }

    void addRam(Ram ram)
    {
        this->ramList.push_back(ram);
    }

    string getNama()
    {
        return this->nama;
    }

    Cpu getCpu()
    {
        return this->cpu;
    }

    vector<Ram> getRamList() // return vector
    {
        return this->ramList;
    }

    HDD gethdd()
    {
        return this->hdd;
    }

    VGA getVga()
    {
        return vga;
    }

    SSD getSsd()
    {
        return ssd;
    }

    string getMerk()
    {
        return this->merk;
    }

    // Getter untuk perangkat eksternal
    Keyboard getKeyboard() { return keyboard; }
    Mouse getMouse() { return mouse; }
    Speaker getSpeaker() { return speaker; }

    ~Komputer()
    {
    }
};

File: Python/Program/Komputer.py
from Cpu import Cpu
from Ram import Ram
from Harddrive import Harddrive
from VGA import VGA
from SSD import SSD
from Keyboard import Keyboard
from Mouse import Mouse
from Speaker import Speaker

class Komputer:
    def __init__(self, nama, merk, cpu, ram_list, harddrive, vga, ssd, keyboard, mouse, speaker):
        self.nama = nama
        self.merk = merk
        self.cpu = cpu
        self.ram_list = ram_list
        self.harddrive = harddrive
        self.vga = vga
        self.ssd = ssd
        self.keyboard = keyboard
        self.mouse = mouse
        self.speaker = speaker

    def get_nama(self):
        return self.nama

    def get_merk(self):
        return self.merk

    def get_cpu(self):
        return self.cpu

    def get_ram_list(self):
        return self.ram_list

    def get_harddrive(self):
        return self.harddrive

    def get_vga(self):
        return self.vga

    def get_ssd(self):
        return self.ssd

    def get_keyboard(self):
        return self.keyboard

    def get_mouse(self):
        return self.mouse

    def get_speaker(self):
        return self.speaker